contracts/wawaka/interpreter-test/interpreter-test.cpp

```diff
@@ -204,6 +204,71 @@ bool rsa_test(const Message& msg, const Environment& env, Response& rsp)
     return rsp.value(v, false);
 }
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// NAME: hash_test
+//
+// test the hash and hmac functions. for the moment this only tests
+// the 256 bit versions.
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+bool hash_test(const Message& msg, const Environment& env, Response& rsp)
+{
+    ww::types::ByteArray hash;
+    std::string encoded_hash;
+
+    ww::types::ByteArray hmac;
+    std::string encoded_hmac;
+
+    const ww::types::ByteArray hmackey{4, 6, 8, 5, 1, 2, 3, 4, 3, 4, 7, 8, 9, 7, 8, 0};
+    const ww::types::ByteArray hmackey_badkey{0, 6, 8, 5, 1, 2, 3, 4, 3, 4, 7, 8, 9, 7, 8, 0};
+    const std::string test_msg_str("Proof of Elapsed Time");
+    const std::string test_msg_bad_str("Proof of Elapsed Time 2");
+    const std::string expected_hmac("mO+yrlHk5HH1vyDlKuSjhTgWR0Y9Iqv1JlZW+pKDwWk=");
+    const std::string expected_hash("43fTaEjBzvug9rf0RRU6anIHfgdoqNjQ/dy/jzcVcAk=");
+
+    ww::types::ByteArray test_msg(test_msg_str.begin(), test_msg_str.end());
+    ww::types::ByteArray test_msg_bad(test_msg_bad_str.begin(), test_msg_bad_str.end());
+
+    // hash
+    if (! ww::crypto::crypto_hash(test_msg, hash))
+        return rsp.error("failed to compute hash");
+    if (! ww::crypto::b64_encode(hash, encoded_hash))
+        return rsp.error("failed to encode hash");
+    if (encoded_hash != expected_hash)
+        return rsp.error("failed to compute the correct hash");
+
+    if (! ww::crypto::crypto_hash(test_msg_bad, hash))
+        return rsp.error("failed to compute hash");
+    if (! ww::crypto::b64_encode(hash, encoded_hash))
+        return rsp.error("failed to encode hash");
+    if (encoded_hash == expected_hash)
```

Comment:
Is this check testing for a collision? Seems odd to do that with a single bad input. Ditto below.

Reply:
More to make sure that the hash function isn't just returning a static value. The full test suite for the functions is in the common library. This is really just checking to make sure the connection to those functions is correct.